src/automations/types.ts

```diff
@@ -0,0 +1,79 @@
+export type AutomationStepAction = "cancel" | "delay" | "send" | "send-list";
+
+export interface IAutomationStep {
+  action: AutomationStepAction;
+  if?: string;
+  ref?: string;
+}
+
+export interface IAutomationCancelStep extends IAutomationStep {
+  action: "cancel";
+  cancelation_token?: string;
+}
+
+export interface IAutomationDelayStep extends IAutomationStep {
+  action: "delay";
+  duration?: string;
+  until?: string;
+}
+
+export interface IAutomationSendStep extends IAutomationStep {
+  action: "send";
+  brand?: string;
+  data?: object;
+  override?: object;
+  profile?: object;
+  recipient?: string;
+  template?: string;
+}
+
+export interface IAutomationSendListStep extends IAutomationStep {
+  action: "send-list";
+  brand?: string;
+  data?: object;
```

Comment:
let's replace object with `Record<string, unknown>` object is still too big of type and potentially can hold almost anything that can be serializable?